pkg/storage/replica.go

```diff
@@ -2715,7 +2723,10 @@ func (r *Replica) handleRaftReadyRaftMuLocked(
 	writer.Close()
 	// Synchronously commit the batch with the Raft log entries and Raft hard
 	// state as we're promising not to lose this data.
-	if err := batch.Commit(syncRaftLog); err != nil {
+	needSync := len(rd.Entries) > 0 ||
```

Comment:
I'd be much less scared about future breakages if we could just import etcd's function. Want to see if they'd be ok making their `mustSync` method public?